python/tests/test_run_config.py

```diff
@@ -48,3 +50,23 @@ async def test_load_parametrized_data_config(set_temporary_env_vars):
                 },
             ],
         }
+
+@pytest.mark.xfail
```

Comment:
Why are we adding a test we expect to fail?

Reply:
As mentioned by @Ankush-lastmile [here](https://github.com/lastmile-ai/aiconfig/pull/923#discussion_r1472205356). There have been changes to OpenAI model paeser as such new way to for Mocking OpenAI API is required. 

The new test introduced works fines based on the earlier mocking approach. Unless a mocking strategy is implemented we would have to use the work around as used here as well https://github.com/sp6370/aiconfig/blob/9617bd9d54e3bff3bb7eef2889af9f30adf39602/python/tests/test_run_config.py#L11.

Reply:
I think it's good to add, we can edit it later

Reply:
I have e-enabled the test.